core/templates/pages/contributor-dashboard-page/modal-templates/certificate-download-modal.component.ts

```diff
@@ -0,0 +1,329 @@
+      linePosition += 40;
+      ctx.font = '24px Roboto';
+      ctx.fillStyle = '#8F9899';
+      ctx.fillText('SIGNATURE', this.SIGNATURE_BASE_COORDINATE, linePosition);
+      ctx.fillText('DATE', this.DATE_BASE_COORDINATE, linePosition);
+
+      // Create an HTML link and clicks on it to download.
+      const link = document.createElement('a');
+      link.download = 'certificate.jpeg';
+      link.href = canvas.toDataURL();
+      link.click();
+    };
+  }
+
+  fillTranslationSubmitterCertificateContent(
+      linePosition: number,
+      ctx: CanvasRenderingContext2D,
+      response: ContributorCertificateResponse
+  ): number {
+    ctx.fillText(
+      'for their dedication and time in translating Oppia\'s basic maths ' +
+      'lessons to ' + response.language,
+      this.CERTIFICATE_MID_POINT,
+      linePosition
+    );
+    // Increase y coordinate by 40.
+    linePosition += 40;
+
+    ctx.fillText(
+      'which will help our ' + response.language + '-speaking learners ' +
+      'better understand the lessons.',
+      this.CERTIFICATE_MID_POINT,
+      linePosition
+    );
+    // Increase y coordinate by 80.
+    linePosition += 80;
+
+    ctx.fillText(
+      `This certificate confirms that ${this.username} has contributed ` +
+      response.contribution_hours + ' hours worth of',
+      this.CERTIFICATE_MID_POINT,
+      linePosition
+    );
+    // Increase y coordinate by 40.
+    linePosition += 40;
+
+    ctx.fillText(
+      `translations from ${response.from_date} to ${response.to_date}.`,
+      this.CERTIFICATE_MID_POINT,
+      860
+    );
+    // Increase y coordinate by 40.
+    linePosition += 100;
+
+    return linePosition;
+  }
+
+  fillQuestionSubmitterCertificateContent(
+      linePosition: number,
+      ctx: CanvasRenderingContext2D,
+      response: ContributorCertificateResponse
+  ): number {
+    ctx.fillText(
+      'for their dedication and time in contributing practice questions ' +
+      'to Oppia\'s',
+      this.CERTIFICATE_MID_POINT,
+      linePosition
+    );
+    // Increase y coordinate by 40.
+    linePosition += 40;
+
+    ctx.fillText(
+      'Math Classroom, which supports our mission of improving',
+      this.CERTIFICATE_MID_POINT,
+      linePosition
+    );
+    // Increase y coordinate by 40.
+    linePosition += 40;
+
+    ctx.fillText(
+      'access to quality education.',
+      this.CERTIFICATE_MID_POINT,
+      linePosition
+    );
+    // Increase y coordinate by 80.
+    linePosition += 80;
+
+    ctx.fillText(
+      `This certificate confirms that ${this.username} has contributed ` +
+      response.contribution_hours + ' hours',
+      this.CERTIFICATE_MID_POINT,
+      linePosition
+    );
+    // Increase y coordinate by 40.
+    linePosition += 40;
+
+    ctx.fillText(
+      `to Oppia from ${response.from_date} to ${response.to_date}.`,
+      this.CERTIFICATE_MID_POINT,
+      linePosition
+    );
+    // Increase y coordinate by 40.
+    linePosition += 40;
```

Comment:
Ditto in the functions above.

Reply:
I removed `fillTranslationSubmitterCertificateContent` `fillQuestionSubmitterCertificateContent` functions and made one single function to fill the dynamic content. But rest of the parts can not be done through a loop because various parts have different configurations of the canvas.